java/code/src/com/suse/manager/reactor/messaging/LibvirtEngineMessage.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2018 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.reactor.messaging;
+
+import com.redhat.rhn.common.messaging.EventMessage;
+
+import com.google.gson.JsonElement;
+import com.suse.salt.netapi.event.EngineEvent;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Libvirt_events engine message to handle
+ */
+public abstract class LibvirtEngineMessage implements EventMessage {
```

Comment:
Since this is an abstract class, should we name it as 'AbstractXXX'?

Reply:
Sure we can if that is the local convention

Reply:
It's not enforced:

```
spacewalk % rg "abstract class" -l | grep -vi abstract | wc -l
121
```

but I'd say it's a good thing. Up to you if you want to adhere or not.

Reply:
OK, I changed the name.